Rakit\Validation\RuleNotFoundException: Validator 'numberic' is not registered in C:\laragon\www\lab1php2\vendor\rakit\validation\src\Validator.php:103
Stack trace:
#0 [internal function]: Rakit\Validation\Validator->__invoke('numberic')
#1 C:\laragon\www\lab1php2\vendor\rakit\validation\src\Validation.php(494): call_user_func_array(Object(Rakit\Validation\Validator), Array)
#2 C:\laragon\www\lab1php2\vendor\rakit\validation\src\Validation.php(71): Rakit\Validation\Validation->resolveRules(Array)
#3 C:\laragon\www\lab1php2\vendor\rakit\validation\src\Validation.php(58): Rakit\Validation\Validation->addAttribute('price', Array)
#4 C:\laragon\www\lab1php2\vendor\rakit\validation\src\Validator.php(83): Rakit\Validation\Validation->__construct(Object(Rakit\Validation\Validator), Array, Array, Array)
#5 C:\laragon\www\lab1php2\app\Controller.php(10): Rakit\Validation\Validator->make(Array, Array)
#6 C:\laragon\www\lab1php2\app\Controllers\Admin\ProductController.php(43): App\Controller->validate(Object(Rakit\Validation\Validator), Array, Array)
#7 [internal function]: App\Controllers\Admin\ProductController->store()
#8 C:\laragon\www\lab1php2\vendor\bramus\router\src\Bramus\Router\Router.php(482): call_user_func_array(Array, Array)
#9 C:\laragon\www\lab1php2\vendor\bramus\router\src\Bramus\Router\Router.php(440): Bramus\Router\Router->invoke('App\\Controllers...', Array)
#10 C:\laragon\www\lab1php2\vendor\bramus\router\src\Bramus\Router\Router.php(287): Bramus\Router\Router->handle(Array, true)
#11 C:\laragon\www\lab1php2\routers\index.php(11): Bramus\Router\Router->run()
#12 C:\laragon\www\lab1php2\index.php(9): require('C:\\laragon\\www\\...')
#13 {main}Rakit\Validation\RuleNotFoundException: Validator 'file_uploaded' is not registered in C:\laragon\www\lab1php2\vendor\rakit\validation\src\Validator.php:103
Stack trace:
#0 [internal function]: Rakit\Validation\Validator->__invoke('file_uploaded', '0', '2048K', 'png', 'jpg', 'jpeg', 'webp')
#1 C:\laragon\www\lab1php2\vendor\rakit\validation\src\Validation.php(494): call_user_func_array(Object(Rakit\Validation\Validator), Array)
#2 C:\laragon\www\lab1php2\vendor\rakit\validation\src\Validation.php(71): Rakit\Validation\Validation->resolveRules(Array)
#3 C:\laragon\www\lab1php2\vendor\rakit\validation\src\Validation.php(58): Rakit\Validation\Validation->addAttribute('img_thumbnail', 'nullable|file_u...')
#4 C:\laragon\www\lab1php2\vendor\rakit\validation\src\Validator.php(83): Rakit\Validation\Validation->__construct(Object(Rakit\Validation\Validator), Array, 'nullable|file_u...', Array)
#5 C:\laragon\www\lab1php2\app\Controller.php(10): Rakit\Validation\Validator->make(Array, Array)
#6 C:\laragon\www\lab1php2\app\Controllers\Admin\ProductController.php(43): App\Controller->validate(Object(Rakit\Validation\Validator), Array, Array)
#7 [internal function]: App\Controllers\Admin\ProductController->store()
#8 C:\laragon\www\lab1php2\vendor\bramus\router\src\Bramus\Router\Router.php(482): call_user_func_array(Array, Array)
#9 C:\laragon\www\lab1php2\vendor\bramus\router\src\Bramus\Router\Router.php(440): Bramus\Router\Router->invoke('App\\Controllers...', Array)
#10 C:\laragon\www\lab1php2\vendor\bramus\router\src\Bramus\Router\Router.php(287): Bramus\Router\Router->handle(Array, true)
#11 C:\laragon\www\lab1php2\routers\index.php(11): Bramus\Router\Router->run()
#12 C:\laragon\www\lab1php2\index.php(9): require('C:\\laragon\\www\\...')
#13 {main}PDOException: SQLSTATE[HY000]: General error: 1366 Incorrect integer value: 'Chon Danh muc' for column 'category_id' at row 1 in C:\laragon\www\lab1php2\vendor\doctrine\dbal\src\Driver\PDO\Statement.php:55
Stack trace:
#0 C:\laragon\www\lab1php2\vendor\doctrine\dbal\src\Driver\PDO\Statement.php(55): PDOStatement->execute()
#1 C:\laragon\www\lab1php2\vendor\doctrine\dbal\src\Connection.php(882): Doctrine\DBAL\Driver\PDO\Statement->execute()
#2 C:\laragon\www\lab1php2\vendor\doctrine\dbal\src\Connection.php(518): Doctrine\DBAL\Connection->executeStatement('INSERT INTO pro...', Array, Array)
#3 C:\laragon\www\lab1php2\app\Model.php(79): Doctrine\DBAL\Connection->insert('products', Array)
#4 C:\laragon\www\lab1php2\app\Controllers\Admin\ProductController.php(84): App\Model->insert(Array)
#5 [internal function]: App\Controllers\Admin\ProductController->store()
#6 C:\laragon\www\lab1php2\vendor\bramus\router\src\Bramus\Router\Router.php(482): call_user_func_array(Array, Array)
#7 C:\laragon\www\lab1php2\vendor\bramus\router\src\Bramus\Router\Router.php(440): Bramus\Router\Router->invoke('App\\Controllers...', Array)
#8 C:\laragon\www\lab1php2\vendor\bramus\router\src\Bramus\Router\Router.php(287): Bramus\Router\Router->handle(Array, true)
#9 C:\laragon\www\lab1php2\routers\index.php(11): Bramus\Router\Router->run()
#10 C:\laragon\www\lab1php2\index.php(9): require('C:\\laragon\\www\\...')
#11 {main}

Next Doctrine\DBAL\Driver\PDO\Exception: SQLSTATE[HY000]: General error: 1366 Incorrect integer value: 'Chon Danh muc' for column 'category_id' at row 1 in C:\laragon\www\lab1php2\vendor\doctrine\dbal\src\Driver\PDO\Exception.php:24
Stack trace:
#0 C:\laragon\www\lab1php2\vendor\doctrine\dbal\src\Driver\PDO\Statement.php(57): Doctrine\DBAL\Driver\PDO\Exception::new(Object(PDOException))
#1 C:\laragon\www\lab1php2\vendor\doctrine\dbal\src\Connection.php(882): Doctrine\DBAL\Driver\PDO\Statement->execute()
#2 C:\laragon\www\lab1php2\vendor\doctrine\dbal\src\Connection.php(518): Doctrine\DBAL\Connection->executeStatement('INSERT INTO pro...', Array, Array)
#3 C:\laragon\www\lab1php2\app\Model.php(79): Doctrine\DBAL\Connection->insert('products', Array)
#4 C:\laragon\www\lab1php2\app\Controllers\Admin\ProductController.php(84): App\Model->insert(Array)
#5 [internal function]: App\Controllers\Admin\ProductController->store()
#6 C:\laragon\www\lab1php2\vendor\bramus\router\src\Bramus\Router\Router.php(482): call_user_func_array(Array, Array)
#7 C:\laragon\www\lab1php2\vendor\bramus\router\src\Bramus\Router\Router.php(440): Bramus\Router\Router->invoke('App\\Controllers...', Array)
#8 C:\laragon\www\lab1php2\vendor\bramus\router\src\Bramus\Router\Router.php(287): Bramus\Router\Router->handle(Array, true)
#9 C:\laragon\www\lab1php2\routers\index.php(11): Bramus\Router\Router->run()
#10 C:\laragon\www\lab1php2\index.php(9): require('C:\\laragon\\www\\...')
#11 {main}

Next Doctrine\DBAL\Exception\DriverException: An exception occurred while executing a query: SQLSTATE[HY000]: General error: 1366 Incorrect integer value: 'Chon Danh muc' for column 'category_id' at row 1 in C:\laragon\www\lab1php2\vendor\doctrine\dbal\src\Driver\API\MySQL\ExceptionConverter.php:91
Stack trace:
#0 C:\laragon\www\lab1php2\vendor\doctrine\dbal\src\Connection.php(1414): Doctrine\DBAL\Driver\API\MySQL\ExceptionConverter->convert(Object(Doctrine\DBAL\Driver\PDO\Exception), Object(Doctrine\DBAL\Query))
#1 C:\laragon\www\lab1php2\vendor\doctrine\dbal\src\Connection.php(1356): Doctrine\DBAL\Connection->handleDriverException(Object(Doctrine\DBAL\Driver\PDO\Exception), Object(Doctrine\DBAL\Query))
#2 C:\laragon\www\lab1php2\vendor\doctrine\dbal\src\Connection.php(888): Doctrine\DBAL\Connection->convertExceptionDuringQuery(Object(Doctrine\DBAL\Driver\PDO\Exception), 'INSERT INTO pro...', Array, Array)
#3 C:\laragon\www\lab1php2\vendor\doctrine\dbal\src\Connection.php(518): Doctrine\DBAL\Connection->executeStatement('INSERT INTO pro...', Array, Array)
#4 C:\laragon\www\lab1php2\app\Model.php(79): Doctrine\DBAL\Connection->insert('products', Array)
#5 C:\laragon\www\lab1php2\app\Controllers\Admin\ProductController.php(84): App\Model->insert(Array)
#6 [internal function]: App\Controllers\Admin\ProductController->store()
#7 C:\laragon\www\lab1php2\vendor\bramus\router\src\Bramus\Router\Router.php(482): call_user_func_array(Array, Array)
#8 C:\laragon\www\lab1php2\vendor\bramus\router\src\Bramus\Router\Router.php(440): Bramus\Router\Router->invoke('App\\Controllers...', Array)
#9 C:\laragon\www\lab1php2\vendor\bramus\router\src\Bramus\Router\Router.php(287): Bramus\Router\Router->handle(Array, true)
#10 C:\laragon\www\lab1php2\routers\index.php(11): Bramus\Router\Router->run()
#11 C:\laragon\www\lab1php2\index.php(9): require('C:\\laragon\\www\\...')
#12 {main}PDOException: SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '1' for key 'products.category_id' in C:\laragon\www\lab1php2\vendor\doctrine\dbal\src\Driver\PDO\Statement.php:55
Stack trace:
#0 C:\laragon\www\lab1php2\vendor\doctrine\dbal\src\Driver\PDO\Statement.php(55): PDOStatement->execute()
#1 C:\laragon\www\lab1php2\vendor\doctrine\dbal\src\Connection.php(882): Doctrine\DBAL\Driver\PDO\Statement->execute()
#2 C:\laragon\www\lab1php2\vendor\doctrine\dbal\src\Connection.php(518): Doctrine\DBAL\Connection->executeStatement('INSERT INTO pro...', Array, Array)
#3 C:\laragon\www\lab1php2\app\Model.php(79): Doctrine\DBAL\Connection->insert('products', Array)
#4 C:\laragon\www\lab1php2\app\Controllers\Admin\ProductController.php(83): App\Model->insert(Array)
#5 [internal function]: App\Controllers\Admin\ProductController->store()
#6 C:\laragon\www\lab1php2\vendor\bramus\router\src\Bramus\Router\Router.php(482): call_user_func_array(Array, Array)
#7 C:\laragon\www\lab1php2\vendor\bramus\router\src\Bramus\Router\Router.php(440): Bramus\Router\Router->invoke('App\\Controllers...', Array)
#8 C:\laragon\www\lab1php2\vendor\bramus\router\src\Bramus\Router\Router.php(287): Bramus\Router\Router->handle(Array, true)
#9 C:\laragon\www\lab1php2\routers\index.php(11): Bramus\Router\Router->run()
#10 C:\laragon\www\lab1php2\index.php(9): require('C:\\laragon\\www\\...')
#11 {main}

Next Doctrine\DBAL\Driver\PDO\Exception: SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '1' for key 'products.category_id' in C:\laragon\www\lab1php2\vendor\doctrine\dbal\src\Driver\PDO\Exception.php:24
Stack trace:
#0 C:\laragon\www\lab1php2\vendor\doctrine\dbal\src\Driver\PDO\Statement.php(57): Doctrine\DBAL\Driver\PDO\Exception::new(Object(PDOException))
#1 C:\laragon\www\lab1php2\vendor\doctrine\dbal\src\Connection.php(882): Doctrine\DBAL\Driver\PDO\Statement->execute()
#2 C:\laragon\www\lab1php2\vendor\doctrine\dbal\src\Connection.php(518): Doctrine\DBAL\Connection->executeStatement('INSERT INTO pro...', Array, Array)
#3 C:\laragon\www\lab1php2\app\Model.php(79): Doctrine\DBAL\Connection->insert('products', Array)
#4 C:\laragon\www\lab1php2\app\Controllers\Admin\ProductController.php(83): App\Model->insert(Array)
#5 [internal function]: App\Controllers\Admin\ProductController->store()
#6 C:\laragon\www\lab1php2\vendor\bramus\router\src\Bramus\Router\Router.php(482): call_user_func_array(Array, Array)
#7 C:\laragon\www\lab1php2\vendor\bramus\router\src\Bramus\Router\Router.php(440): Bramus\Router\Router->invoke('App\\Controllers...', Array)
#8 C:\laragon\www\lab1php2\vendor\bramus\router\src\Bramus\Router\Router.php(287): Bramus\Router\Router->handle(Array, true)
#9 C:\laragon\www\lab1php2\routers\index.php(11): Bramus\Router\Router->run()
#10 C:\laragon\www\lab1php2\index.php(9): require('C:\\laragon\\www\\...')
#11 {main}

Next Doctrine\DBAL\Exception\UniqueConstraintViolationException: An exception occurred while executing a query: SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '1' for key 'products.category_id' in C:\laragon\www\lab1php2\vendor\doctrine\dbal\src\Driver\API\MySQL\ExceptionConverter.php:49
Stack trace:
#0 C:\laragon\www\lab1php2\vendor\doctrine\dbal\src\Connection.php(1414): Doctrine\DBAL\Driver\API\MySQL\ExceptionConverter->convert(Object(Doctrine\DBAL\Driver\PDO\Exception), Object(Doctrine\DBAL\Query))
#1 C:\laragon\www\lab1php2\vendor\doctrine\dbal\src\Connection.php(1356): Doctrine\DBAL\Connection->handleDriverException(Object(Doctrine\DBAL\Driver\PDO\Exception), Object(Doctrine\DBAL\Query))
#2 C:\laragon\www\lab1php2\vendor\doctrine\dbal\src\Connection.php(888): Doctrine\DBAL\Connection->convertExceptionDuringQuery(Object(Doctrine\DBAL\Driver\PDO\Exception), 'INSERT INTO pro...', Array, Array)
#3 C:\laragon\www\lab1php2\vendor\doctrine\dbal\src\Connection.php(518): Doctrine\DBAL\Connection->executeStatement('INSERT INTO pro...', Array, Array)
#4 C:\laragon\www\lab1php2\app\Model.php(79): Doctrine\DBAL\Connection->insert('products', Array)
#5 C:\laragon\www\lab1php2\app\Controllers\Admin\ProductController.php(83): App\Model->insert(Array)
#6 [internal function]: App\Controllers\Admin\ProductController->store()
#7 C:\laragon\www\lab1php2\vendor\bramus\router\src\Bramus\Router\Router.php(482): call_user_func_array(Array, Array)
#8 C:\laragon\www\lab1php2\vendor\bramus\router\src\Bramus\Router\Router.php(440): Bramus\Router\Router->invoke('App\\Controllers...', Array)
#9 C:\laragon\www\lab1php2\vendor\bramus\router\src\Bramus\Router\Router.php(287): Bramus\Router\Router->handle(Array, true)
#10 C:\laragon\www\lab1php2\routers\index.php(11): Bramus\Router\Router->run()
#11 C:\laragon\www\lab1php2\index.php(9): require('C:\\laragon\\www\\...')
#12 {main}PDOException: SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '1' for key 'products.category_id' in C:\laragon\www\lab1php2\vendor\doctrine\dbal\src\Driver\PDO\Statement.php:55
Stack trace:
#0 C:\laragon\www\lab1php2\vendor\doctrine\dbal\src\Driver\PDO\Statement.php(55): PDOStatement->execute()
#1 C:\laragon\www\lab1php2\vendor\doctrine\dbal\src\Connection.php(882): Doctrine\DBAL\Driver\PDO\Statement->execute()
#2 C:\laragon\www\lab1php2\vendor\doctrine\dbal\src\Connection.php(518): Doctrine\DBAL\Connection->executeStatement('INSERT INTO pro...', Array, Array)
#3 C:\laragon\www\lab1php2\app\Model.php(79): Doctrine\DBAL\Connection->insert('products', Array)
#4 C:\laragon\www\lab1php2\app\Controllers\Admin\ProductController.php(83): App\Model->insert(Array)
#5 [internal function]: App\Controllers\Admin\ProductController->store()
#6 C:\laragon\www\lab1php2\vendor\bramus\router\src\Bramus\Router\Router.php(482): call_user_func_array(Array, Array)
#7 C:\laragon\www\lab1php2\vendor\bramus\router\src\Bramus\Router\Router.php(440): Bramus\Router\Router->invoke('App\\Controllers...', Array)
#8 C:\laragon\www\lab1php2\vendor\bramus\router\src\Bramus\Router\Router.php(287): Bramus\Router\Router->handle(Array, true)
#9 C:\laragon\www\lab1php2\routers\index.php(11): Bramus\Router\Router->run()
#10 C:\laragon\www\lab1php2\index.php(9): require('C:\\laragon\\www\\...')
#11 {main}

Next Doctrine\DBAL\Driver\PDO\Exception: SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '1' for key 'products.category_id' in C:\laragon\www\lab1php2\vendor\doctrine\dbal\src\Driver\PDO\Exception.php:24
Stack trace:
#0 C:\laragon\www\lab1php2\vendor\doctrine\dbal\src\Driver\PDO\Statement.php(57): Doctrine\DBAL\Driver\PDO\Exception::new(Object(PDOException))
#1 C:\laragon\www\lab1php2\vendor\doctrine\dbal\src\Connection.php(882): Doctrine\DBAL\Driver\PDO\Statement->execute()
#2 C:\laragon\www\lab1php2\vendor\doctrine\dbal\src\Connection.php(518): Doctrine\DBAL\Connection->executeStatement('INSERT INTO pro...', Array, Array)
#3 C:\laragon\www\lab1php2\app\Model.php(79): Doctrine\DBAL\Connection->insert('products', Array)
#4 C:\laragon\www\lab1php2\app\Controllers\Admin\ProductController.php(83): App\Model->insert(Array)
#5 [internal function]: App\Controllers\Admin\ProductController->store()
#6 C:\laragon\www\lab1php2\vendor\bramus\router\src\Bramus\Router\Router.php(482): call_user_func_array(Array, Array)
#7 C:\laragon\www\lab1php2\vendor\bramus\router\src\Bramus\Router\Router.php(440): Bramus\Router\Router->invoke('App\\Controllers...', Array)
#8 C:\laragon\www\lab1php2\vendor\bramus\router\src\Bramus\Router\Router.php(287): Bramus\Router\Router->handle(Array, true)
#9 C:\laragon\www\lab1php2\routers\index.php(11): Bramus\Router\Router->run()
#10 C:\laragon\www\lab1php2\index.php(9): require('C:\\laragon\\www\\...')
#11 {main}

Next Doctrine\DBAL\Exception\UniqueConstraintViolationException: An exception occurred while executing a query: SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '1' for key 'products.category_id' in C:\laragon\www\lab1php2\vendor\doctrine\dbal\src\Driver\API\MySQL\ExceptionConverter.php:49
Stack trace:
#0 C:\laragon\www\lab1php2\vendor\doctrine\dbal\src\Connection.php(1414): Doctrine\DBAL\Driver\API\MySQL\ExceptionConverter->convert(Object(Doctrine\DBAL\Driver\PDO\Exception), Object(Doctrine\DBAL\Query))
#1 C:\laragon\www\lab1php2\vendor\doctrine\dbal\src\Connection.php(1356): Doctrine\DBAL\Connection->handleDriverException(Object(Doctrine\DBAL\Driver\PDO\Exception), Object(Doctrine\DBAL\Query))
#2 C:\laragon\www\lab1php2\vendor\doctrine\dbal\src\Connection.php(888): Doctrine\DBAL\Connection->convertExceptionDuringQuery(Object(Doctrine\DBAL\Driver\PDO\Exception), 'INSERT INTO pro...', Array, Array)
#3 C:\laragon\www\lab1php2\vendor\doctrine\dbal\src\Connection.php(518): Doctrine\DBAL\Connection->executeStatement('INSERT INTO pro...', Array, Array)
#4 C:\laragon\www\lab1php2\app\Model.php(79): Doctrine\DBAL\Connection->insert('products', Array)
#5 C:\laragon\www\lab1php2\app\Controllers\Admin\ProductController.php(83): App\Model->insert(Array)
#6 [internal function]: App\Controllers\Admin\ProductController->store()
#7 C:\laragon\www\lab1php2\vendor\bramus\router\src\Bramus\Router\Router.php(482): call_user_func_array(Array, Array)
#8 C:\laragon\www\lab1php2\vendor\bramus\router\src\Bramus\Router\Router.php(440): Bramus\Router\Router->invoke('App\\Controllers...', Array)
#9 C:\laragon\www\lab1php2\vendor\bramus\router\src\Bramus\Router\Router.php(287): Bramus\Router\Router->handle(Array, true)
#10 C:\laragon\www\lab1php2\routers\index.php(11): Bramus\Router\Router->run()
#11 C:\laragon\www\lab1php2\index.php(9): require('C:\\laragon\\www\\...')
#12 {main}